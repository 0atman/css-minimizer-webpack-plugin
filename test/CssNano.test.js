const CssnanoPlugin = require('../src/index');
const { createCompiler, compile } = require('./helpers');
const MiniCssExtractPlugin = require('mini-css-extract-plugin');

describe('CssnanoPlugin', () => {
  jest.setTimeout(30000);

  it('should work with assets using querystring', () => {
    const config = {
      devtool: 'sourcemap',
      entry: {
        entry: `${__dirname}/fixtures/foo.css`
      },
      plugins: [
        new MiniCssExtractPlugin({
          filename: '[name].css?v=[chunkhash]',
          chunkFilename: '[id].[name].css?v=[chunkhash]'
        })
      ]
    };

    const compiler = createCompiler(config);
    new CssnanoPlugin({
      sourceMap: true
    }).apply(compiler);

    return compile(compiler).then(stats => {
      expect(stats.compilation.errors).toEqual([]);
      expect(stats.compilation.warnings).toEqual([]);

      for (const file in stats.compilation.assets) {
        if (/\.js/.test(file)) continue;
        expect(stats.compilation.assets[file].source()).toMatchSnapshot(file);
      }
    });
  });
});
