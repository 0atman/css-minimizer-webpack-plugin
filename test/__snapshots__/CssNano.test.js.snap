// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CssnanoPlugin should work with assets using querystring: entry.css.map?v=3868ca6c9937500db3d7 1`] = `"{\\"version\\":3,\\"sources\\":[\\"webpack:///./test/fixtures/foo.css\\"],\\"names\\":[],\\"mappings\\":\\"AAAA,KACA,SACA,CACA,EACA,UACA\\",\\"file\\":\\"entry.css?v=3868ca6c9937500db3d7\\",\\"sourcesContent\\":[\\"body {\\\\n  color: red;\\\\n}\\\\na {\\\\n  color: blue;\\\\n}\\"],\\"sourceRoot\\":\\"\\"}"`;

exports[`CssnanoPlugin should work with assets using querystring: entry.css?v=3868ca6c9937500db3d7 1`] = `
"body{color:red}a{color:#00f}
/*# sourceMappingURL=entry.css.map?v=3868ca6c9937500db3d7*/"
`;
